packages/plugin-lighthouse/src/lib/lighthouse-plugin.integration.test.ts

```diff
@@ -1,12 +1,63 @@
-import { expect } from 'vitest';
-import { pluginConfigSchema } from '@code-pushup/models';
-import { lighthousePlugin } from './lighthouse-plugin';
+import { rm } from 'node:fs/promises';
+import { join } from 'node:path';
+import { afterEach, expect } from 'vitest';
+import { AuditOutput, pluginConfigSchema } from '@code-pushup/models';
+import { getLogMessages } from '@code-pushup/test-utils';
+import { ui } from '@code-pushup/utils';
+import { createRunnerFunction, lighthousePlugin } from './lighthouse-plugin';
+
+const lighthousePluginTestFolder = join('tmp', 'plugin-lighthouse');
 
 describe('lighthousePlugin', () => {
   it('should create valid plugin config', () => {
-    const pluginConfig = lighthousePlugin('https://code-pushup-portal.com');
+    const pluginConfig = lighthousePlugin('https://www.google.com/');
     expect(() => pluginConfigSchema.parse(pluginConfig)).not.toThrow();
     expect(pluginConfig.audits.length).toBeGreaterThan(100);
     expect(pluginConfig.groups).toHaveLength(5);
   });
 });
+
+describe('createRunnerFunction', () => {
+  const getRunnerTestFolder = join(lighthousePluginTestFolder, 'get-runner');
+
+  afterEach(async () => {
+    await rm(getRunnerTestFolder, { recursive: true, force: true });
+  });
+
+  afterAll(async () => {
+    await rm(lighthousePluginTestFolder, { recursive: true, force: true });
+  });
+
+  it('should create and execute runner correctly', async () => {
+    const runner = createRunnerFunction('https://www.google.com/', {
+      // onlyAudits is used to reduce test time
+      onlyAudits: ['is-on-https'],
+      outputPath:
+        'tmp/plugin-lighthouse/get-runner/should-create/lh-report.json',
+      chromeFlags: ['--headless=shell'],
+    });
+    await expect(runner(undefined)).resolves.toEqual([
+      expect.objectContaining({
+        slug: 'is-on-https',
+        score: 1,
+        value: 0,
+      } satisfies AuditOutput),
+    ]);
+  });
+
+  it('should log about unsupported precomputedLanternDataPath flag', async () => {
+    const precomputedLanternDataPath = join('path', 'to', 'latern-data-folder');
+    const runner = createRunnerFunction('https://www.google.com/', {
+      precomputedLanternDataPath,
+      // onlyAudits is used to reduce test time
+      onlyAudits: ['is-on-https'],
+      outputPath:
+        'tmp/plugin-lighthouse/get-runner/no-latern-data/lh-report.json',
+      chromeFlags: ['--headless=shell'],
+    });
+    await expect(runner(undefined)).resolves.toBeTruthy();
+    expect(getLogMessages(ui().logger).at(0)).toMatch(
+      `Parsing precomputedLanternDataPath "${precomputedLanternDataPath}" is skipped as not implemented.`,
+    );
+  });
+}, 38_000);
```

Comment:
This is set to 38_000 to fix the flakyness. I documented the how I figured that out in the troubleshooting guide.